### PR TITLE
Import multiple DLC datasets from a folder

### LINKED ISSUE #412: Importing maDLC labels
Hi, 

We have a large set of manually labeled frames used in social DLC. Is there a way to train a model with SLEAP using these labeled frames? It would be a bit of a tall task to re-label all of these frames manually. 

Thanks, 

Rauf

### PR DESCRIPTION
**Pull request by ariematsliah**
Import multiple DLC datasets in a folder.

Assuming a folder structure like this:

```
labeled-data/
    video1/
        my_labels.csv
        img00000.png
        img00001.png
        ...
    video2/
    ...
```

user can select the `'labeled-data' `folder in the dialog, and then all `'*.csv'` files 2 levels below will be imported and merged into a single `Sleap` dataset.

Fixes #412

### CODE REVIEW

File: sleap/gui/commands.py [hunk 1]
```diff
@@ -762,6 +767,46 @@ def ask(context: "CommandContext", params: dict) -> bool:
...
+        dirname = FileDialog.openDir(
+            context.app,
+            dir=None,
+            caption="Select a folder with DeepLabCut datasets...",
+        )
+
+        if len(dirname) == 0:
+            return False
+        params["dirname"] = dirname
+        params["filename"] = "converted_dlc_dataset"
```
[the remark this answers is not in the document]
Comment (ariematsliah) [reply sleap/gui/commands.py]: done

File: sleap/gui/commands.py [hunk 2]
```diff
@@ -762,6 +767,46 @@ def ask(context: "CommandContext", params: dict) -> bool:
...
+                merged_labels = labels
+            else:
+                merged_labels.extend_from(labels, unify=True)
+
+        print("Merged labels:")
+        merged_labels.describe()
+
+        merged_labels.save("{}.slp".format(params["filename"]))
+        merged_labels.save("{}.pkg.slp".format(params["filename"]),
+                           with_images=True)  # comment this out to skip saving with embedded images
```
[the remark this answers is not in the document]
Comment (ariematsliah) [reply sleap/gui/commands.py]: done

File: sleap/gui/commands.py [hunk 3]
```diff
@@ -762,6 +767,46 @@ def ask(context: "CommandContext", params: dict) -> bool:
         return True
 
 
+class ConvertDeepLabCut(AppCommand):
+    @staticmethod
+    def do_action(context: "CommandContext", params: dict):
+
+        csv_files = glob("{}/*/*.csv".format(params["dirname"]))
```
[the remark this answers is not in the document]
Comment (ariematsliah) [reply sleap/gui/commands.py]: done

File: tests/data/dlc_multiple_datasets/video2/dlc_dataset_2.csv [hunk 4]
```diff
@@ -0,0 +1,7 @@
+scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer,Scorer
+individuals,Animal1,Animal1,Animal1,Animal1,Animal1,Animal1,Animal2,Animal2,Animal2,Animal2,Animal2,Animal2
+bodyparts,A,A,B,B,C,C,A,A,B,B,C,C
+coords,x,y,x,y,x,y,x,y,x,y,x,y
+labeled-data/video/img000.png,0,1,2,3,4,5,6,7,8,9,10,11
+labeled-data/video/img001.png,12,13,,,15,16,17,18,,,20,21
+labeled-data/video/img002.png,22,23,24,25,26,27,,,,,,
```
Comment (talmo) [inline tests/data/dlc_multiple_datasets/video2/dlc_dataset_2.csv:7]: Maybe change a couple of these numbers so we can test that we're not just importing the same thing twice?

**Review by talmo — CHANGES_REQUESTED** [review]
Just missing the tests and we're good to go :)

File: tests/data/dlc_multiple_datasets/video2/dlc_dataset_2.csv (hunk 4 above)
Comment (ariematsliah) [reply tests/data/dlc_multiple_datasets/video2/dlc_dataset_2.csv]: Split the 2 files to have a bit more realistic test. One has labels from the first 2 frames, the other one from the last frame. Also added the images for manual verification via UI

**Comment by ariematsliah** [comment]
PTAL

**Pull request merged**